### PR TITLE
optional cephfs

### PR DESCRIPTION
**Pull request by laverya**

### CODE REVIEW

**Review by emosbaugh — CHANGES_REQUESTED** [review]
I think you need to make kurlkinds

File: addons/rook/template/base/install.sh [hunk 1]
```diff
@@ -146,14 +146,30 @@ function rook_cluster_deploy() {
     render_yaml_file_2 "$dst/tmpl-rbd-storageclass.yaml" > "$dst/rbd-storageclass.yaml"
     insert_resources "$dst/kustomization.yaml" rbd-storageclass.yaml
 
+    # conditional cephfs
+    if [ "${ROOK_SHARED_FILESYSTEM_DISABLED}" != "1" ]; then
+        insert_resources "$dst/kustomization.yaml" cephfs-storageclass.yaml
+        insert_resources "$dst/kustomization.yaml" filesystem.yaml
+        insert_patches_strategic_merge "$dst/kustomization.yaml" patches/cephfs-storageclass.yaml
+        insert_patches_strategic_merge "$dst/kustomization.yaml" patches/filesystem.yaml
+        cat << EOF >> "$dst/kustomization.yaml"
```
Comment (emosbaugh) [inline addons/rook/template/base/install.sh:155]: this seems brittle? is it possible to use function insert_patches_(Json6902)?

**Review by emosbaugh — CHANGES_REQUESTED** [review]

File: addons/rook/template/base/install.sh (hunk 1 above)
Comment (laverya) [reply addons/rook/template/base/install.sh:155]: we do not yet have such a function, though I can add one

**Comment by laverya** [comment]
to clarify, I will remove the duplicate `insert_patches_json_6902` function from the addon directory before merging - that's just needed for testgrid

**Review by emosbaugh — APPROVED** [review]

**Pull request merged**